solr/core/src/java/org/apache/solr/request/SolrRequestInfo.java

```diff
@@ -52,35 +56,53 @@
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
   public static SolrRequestInfo getRequestInfo() {
-    return threadLocal.get();
+    if (threadLocal.get().isEmpty()) return null;
```

Comment:
Instead of calling threadLocal.get() twice, can you please just call it once?  This goes for all the other methods in this class too.